lib/edusign/client.rb

```diff
@@ -1,18 +1,24 @@
+require "active_support/configurable"
 require "httparty"
 
 module Edusign
   class Client
+    include ActiveSupport::Configurable
     include HTTParty
     base_uri "https://ext.edusign.fr/v1"
 
     ALREADY_LOCKED_ERROR_MESSAGE = "Course already locked".freeze
     STUDENT_ALREADY_ADDED_TO_COURSE_ERROR_MESSAGE = "Student already in the list"
     LOCKED_ERROR_MESSAGE = "course locked".freeze
 
+    class NoApiKeyError < StandardError; end
+
     class BadGatewayError < StandardError; end
 
-    def initialize(account_api_key:)
+    def initialize(account_api_key: config.account_api_key)
```

Comment:
When is `config.account_api_key` evaluated? At runtime when calling the method? Or at class load? Did you test it already with a `configure` block compared to the solution I gave with `||` ?

Reply:
Fine with me then with the default keyword argument!

Reply:
And it's evaluated at class load 👇 

```ruby
[3] pry(main)> Edusign::Client.config
=> {:account_api_key=>"7a8ed36625ecd9d8cd2fa3962ced043c0d1c8dc117a9121314a3622e5b7f3e92"}
```

But it works when I instantiate a new client
```ruby
[1] pry(main)> Edusign::Client.new
=> #<Edusign::Client:0x000000010fdcd8a8
 @_config={},
 @account_api_key="7a8ed36625ecd9d8cd2fa3962ced043c0d1c8dc117a9121314a3622e5b7f3e92">
```